### DISCUSSION TITLE
Can you suggest me any best book for learning git?

### DISCUSSION THREAD

**Discussion by Mashpy** [community/community#24147] [New to GitHub]
<p>I have basic idea of git. like commit, branch, merge conflict fix. I want to know more about git like rebase, reflog, file management, how git works and how can I contribute popular open source projects. Can you suggest me any good book for me?</p>

**Comment by mashpysays** [comment, marked as answer]
<p>You can follow <a href="http://www.onlinebooksreview.com/articles/best-git-book-for-beginner-professional-programmers-and-open-source-developers#B00QFIA5OC" rel="nofollow noopener">Ry’s Git Tutorial</a> or <a href="http://www.onlinebooksreview.com/articles/best-git-book-for-beginner-professional-programmers-and-open-source-developers#1484200772" rel="nofollow noopener">Pro Git</a> for learning git. Hope these books are helpful for you.</p>

**Comment by osalifu** [comment]
<p>This book helped me a lot. I hope it helps you too…</p>
<p><a href="https://github.s3.amazonaws.com/media/book.pdf" rel="nofollow noopener">https://github.s3.amazonaws.com/media/book.pdf</a></p>

**Comment by lukeMersh** [comment]
<p>Hi there.</p>
<p>I am looking to learn Github, i see you recommend for Git, but would this book be suitable for Github?</p>
<p>many thanks</p>

**Comment by a-a-ron** [comment]
<p>Hey Luke,</p>
<p>The previously mentioned books focus mostly on Git rather than GitHub. If you’re looking for a GitHub specific book, I’d recommend the book <a href="https://www.amazon.com/Introducing-GitHub-Non-Technical-Peter-Bell/dp/1491949740" rel="nofollow noopener">Introducing GitHub: A Non-Technical Guide</a> . It’s a bit out-of-date but the collaboration features and best practices remain. </p>
<p>When it comes to learning GitHub, we believe that working in hands-on projects on GitHub is the best place to learn how it works. As we continue to build more courses on Learning Lab, we hope to bridge the gap of missing content and keep things relevant as you learn in a working Git repository on GitHub. </p>

**Comment by ntaranov** [comment]
<p>I’m currently working on a course on more elaborate Continuous Integration scenarios like</p>
<p>This includes:</p>
<ul>
<li>Creating and merging a PR</li>
<li>The case with concurrent commits to master branch</li>
<li>Reverting changes causing a failure</li>
</ul>
<p>If I overcome current problems with creating a course, you can expect it in a week or so.</p>

**Comment by lucidl** [comment]
<p>I created this <a href="https://github.com/lucidl/git-internals-tdg" rel="noopener nofollow ugc">https://github.com/lucidl/git-internals-tdg</a><br>
learning material, based on <em><a href="https://github.com/pluralsight/git-internals-pdf" rel="noopener nofollow ugc">Git Internals</a></em> by <em>Scott Chacon</em></p>

**Comment by mrnazu** [comment]
use github cheetsheet or read this book "Ry's Git Tutorial"

**Comment by osalifu** [comment]
I have a very nice book. Is pdf. Will look for it and send it to youThanksOsumanu SalifuSent from my iPhoneOn Nov 29, 2022, at 4:55 PM, Samuel ( NAZU ) ***@***.***> wrote:﻿
use github cheetsheet or read this book "Ry's Git Tutorial"

**Comment by WadeStorm05** [comment]
One thing that helped me was treating Git learning like an [ikigai](https://ikigaiibook.com/en) process. You learn best when what you study connects to what you enjoy building and what you actually use day to day. Instead of reading everything end to end, pick a feature you need, like resolving merge conflicts or cleaning commit history, read that section, then immediately try it on a small test repo or an open source project. Also, contributing to beginner friendly open source projects teaches Git faster than any book. You naturally learn pull requests, reviews, rebasing, and collaboration workflows in context, which is hard to fully grasp from theory alone.
